src/software/amazon/ion/IonReader.java

```diff
@@ -172,6 +172,17 @@
     public IonType getType();
 
     /**
+     * Returns the actual type of the Ion <code>int</code> at the current
+     * value.
+     *
+     * @return an {@link IntegerType} denoting the current <code>int</code>
+     * value's Java type, or <code>null</code> if the current value is
+     * <code>null.int</code>.
+     * @see IonInt#getIntegerType()
+     */
+    public IntegerType getIntegerType();
```

Comment:
+1 to eliminating the `IntegerType.None` and using `null`, though I'm sure the non-`null` purists out there will not like that.